scalafix-core/shared/src/main/scala/scalafix/internal/rule/NoUnitInsertion.scala

```diff
@@ -0,0 +1,25 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scalafix.{Patch, SemanticdbIndex}
+import scalafix.rule.{RuleCtx, SemanticRule}
+
+case class NoUnitInsertion(index: SemanticdbIndex)
+    extends SemanticRule(index, "NoUnitInsertion") {
+
+  private[this] def insertUnit(ctx: RuleCtx, t: Term.Apply): Patch =
+    ctx.addRight(t.tokens.init.last, "()")
+
+  override def fix(ctx: RuleCtx): Patch = {
+    val adaptations = index.messages.toIterator.collect {
```

Comment:
This will iterate through all reported messages in the workspace/project on every source file, we should move it outside of the `fix` method.

Reply:
Sure, I'll do the same in NoAutoTupling